velox/exec/tests/utils/TpchQueryBuilder.cpp

```diff
@@ -688,8 +690,82 @@ TpchPlan TpchQueryBuilder::getQ18Plan() const {
   return context;
 }
 
+TpchPlan TpchQueryBuilder::getQ19Plan() const {
+  static const std::string kLineitem = "lineitem";
+  static const std::string kPart = "part";
+  std::vector<std::string> lineitemColumns = {
+      "l_partkey",
+      "l_shipmode",
+      "l_shipinstruct",
+      "l_extendedprice",
+      "l_discount",
+      "l_quantity"};
+  std::vector<std::string> partColumns = {
+      "p_partkey", "p_brand", "p_container", "p_size"};
+
+  auto lineitemSelectedRowType = getRowType(kLineitem, lineitemColumns);
+  const auto& lineitemFileColumns = getFileColumnNames(kLineitem);
+  auto partSelectedRowType = getRowType(kPart, partColumns);
+  const auto& partFileColumns = getFileColumnNames(kPart);
+
+  auto planNodeIdGenerator = std::make_shared<PlanNodeIdGenerator>();
+  core::PlanNodeId lineitemScanNodeId;
+  core::PlanNodeId partScanNodeId;
+
+  std::string shipModeFilter = "l_shipmode IN ('AIR', 'AIR REG')";
+  std::string shipInstructFilter = "(l_shipinstruct = 'DELIVER IN PERSON')";
+  std::string filterExpr =
+      "     ((p_brand = 'Brand#12')"
+      "     AND (l_quantity between 1.0 and 11.0)"
+      "     AND (p_container IN ('SM CASE', 'SM BOX', 'SM PACK', 'SM PKG'))"
+      "     AND (p_size BETWEEN 1 AND 5))"
+      " OR  ((p_brand ='Brand#23')"
+      "     AND (p_container IN ('MED BAG', 'MED BOX', 'MED PKG', 'MED PACK'))"
+      "     AND (l_quantity between 10.0 and 20.0)"
+      "     AND (p_size BETWEEN 1 AND 10))"
+      " OR  ((p_brand = 'Brand#34')"
+      "     AND (p_container IN ('LG CASE', 'LG BOX', 'LG PACK', 'LG PKG'))"
+      "     AND (l_quantity between 20.0 and 30.0)"
+      "     AND (p_size BETWEEN 1 AND 15))";
+
+  auto part = PlanBuilder(planNodeIdGenerator)
+                  .tableScan(kPart, partSelectedRowType, partFileColumns)
+                  .capturePlanNodeId(partScanNodeId)
+                  .planNode();
+
+  auto pool = memory::getDefaultScopedMemoryPool();
```

Comment:
There isn't a need for this local variable. The pool can be constructed in-place in the PlanBuilder constructor.

Reply:
My bad... getDefaultScopedMemoryPool() returns a unique_ptr. Its better to keep it in a local variable.